packages/apps/docs/src/scripts/importChangelogs/index.ts

```diff
@@ -0,0 +1,220 @@
+          if (!currentContent[repo.name]?.content[version.label].isLocked) {
+            content[version.label] = version;
+            currentPosition = VersionPosition.VERSION;
+          } else {
+            content[version.label] =
+              currentContent[repo.name].content[version.label];
+            version = undefined;
+          }
+        }
+
+        if (branch.type === 'heading' && branch.depth === 3) {
+          const value = (branch.children[0] as Text).value;
+          switch (true) {
+            case checkPatchNames(value):
+              currentPosition = VersionPosition.PATCH;
+              break;
+            case checkMinorNames(value):
+              currentPosition = VersionPosition.MINOR;
+              break;
+            default:
+              currentPosition = VersionPosition.MISC;
+              break;
+          }
+        }
+
+        if (branch.type === 'listItem' && version) {
+          const record = createRecord(branch);
+
+          switch (currentPosition) {
+            case VersionPosition.MINOR:
+              content[version.label].minors.push(record);
+              break;
+            case VersionPosition.PATCH:
+              content[version.label].patches.push(record);
+              break;
+            default:
+              content[version.label].miscs.push(record);
+              break;
+          }
+        } else {
+          innerCrawl(branch);
+        }
+      });
+    }
+
+    return { ...repo, content };
+  };
+
+  return innerCrawl;
+};
+
+const createContent = async (repo: IRepo) => {
+  const md: Root = remark.parse(getChangelog(repo));
+
+  const repoCrawler = await crawl(repo);
+  return await repoCrawler(md);
+};
+
+const getContent = async (repos: IRepo[]): Promise<IChangelogComplete> => {
+  const promises = repos.map(createContent);
+
+  const results = await Promise.allSettled(promises);
+
+  const content: IChangelogComplete = getCurrentContent();
+
+  results.forEach((result, idx) => {
+    if (result.status === 'rejected') {
+      errors.push(`${repos[idx].repo} had issues with creating content`);
+    } else {
+      content[result.value.name] = result.value;
+    }
+  });
+
+  return content;
+};
+
+const getRepos = async (repos: IRepo[]): Promise<void> => {
+  const promises = repos.map((repo) => {
+    return clone(repo.repo);
+  });
+
+  const results = await Promise.allSettled(promises);
+
+  results.forEach((result, idx) => {
+    if (result.status === 'rejected') {
+      errors.push(`${repos[idx].repo} had issues with cloning`);
+    }
+  });
+};
+
+export const importChangelogs = async (): Promise<IScriptResult> => {
+  errors.length = 0;
+  success.length = 0;
+
+  await getRepos(REPOS);
+  const content = await getContent(REPOS);
+
+  await getGitHubData(content);
+  enrichContent(content);
+
+  if (!errors.length) {
+    writeContent(content);
+    success.push('Changelogs imported');
+  }
+
+  await runPrettier();
+  console.log({ errors });
```

Comment:
is it intended?

Reply:
yes. it was part of the build scripts. but will keep it out. so I wanted a quick print